### PR TITLE
Remove AT_DISPATCH_COMPLEX_TYPES_AND and AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX

### PR DESCRIPTION
**Pull request by zasdfgbnm**
These two macros only appear in `Dispatch.h`

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
@anjali411 Ready

**Comment by anjali411** [comment]
should we just get rid of these macros then?
cc. @ezyang

**Comment by ezyang** [comment]
Yeah if they're not used by anything (and we don't plan to use them in the future), let's delete them.

**Comment by anjali411** [comment]
@zasdfgbnm Let's get rid of these macros if they are not being used anywhere

**Comment by zasdfgbnm** [comment]
@anjali411 `AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX` is removed.

File: aten/src/ATen/Dispatch.h [hunk 1]
```diff
@@ -414,9 +414,9 @@ inline void deprecated_AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX() {}
   [&] {                                                                     \
```
Comment (anjali411) [inline aten/src/ATen/Dispatch.h:414]: I think we can remove this as well?

Comment (zasdfgbnm) [reply aten/src/ATen/Dispatch.h:414]: Won't it be used in https://github.com/pytorch/pytorch/pull/37791?

Comment (anjali411) [reply aten/src/ATen/Dispatch.h:414]: no

Comment (zasdfgbnm) [reply aten/src/ATen/Dispatch.h]: I removed it

**Review by anjali411 — APPROVED** [review]

**Pull request closed without merging**